gateleen-core/src/main/java/org/swisspush/gateleen/core/http/LocalHttpClientRequest.java

```diff
@@ -473,6 +475,113 @@ public LocalHttpClientRequest(HttpMethod method, String uri, Vertx vertx, Handle
         this.uri = uri;
         this.routingContextHandler = routingContextHandler;
         this.response = response;
+        // this is a mock-connection object which silently ignores (nearly) HTTP/1.1 relevant method calls and never throws exceptions
+        // we need this to be able to set appropriate exception- and close-handlers in e.g. Gateleen's Forwarder (gateleen-routing)
+        this.connection = new HttpConnection() {
```

Comment:
I suggest to split this up as follows:

1) Introduce a class with its single purpose to throw no matter which method gets called:

```
class FailFastHttpConnection implements HttpConnection {
    
    private static final String msg = "This (mocked) connection doesn't support that call.";

    // Implement ALL methods to simply throw an exception.
    @Override
    public HttpConnection goAway(long errorCode, int lastStreamId, Buffer debugData) {
        throw new UnsupportedOperationException( msg );
    }
}
```

2) Then to create our instance:

```
    this.connection = new FailFastHttpConnection(){
        // Now here we only need to implement special cases because all other
        // methods we don't care about already throw in our super class.
        @Override
        public HttpConnection closeHandler(Handler<Void> handler) {
            return this;
        }
    };
```

This IMO would be:

1) Much simpler to read and understand.
2) Our extra class can simply be reused.
3) No distracting code bloat in our logic.